From 3197660f82762c888064b39165cdf6c83974f3d3 Mon Sep 17 00:00:00 2001
From: Dong Aisheng <aisheng.dong@nxp.com>
Date: Fri, 24 Nov 2017 16:48:12 +0800
Subject: [PATCH 2960/5242] MLK-17074-5 soc: imx8: pm-domains: add multiple
 states

commit  90a9f2bf1f60d98016f98df5f5531be5ba26df90 from
https://source.codeaurora.org/external/imx/linux-imx.git

MX8 power domain supports two low power modes: LP and OFF.
So adding them accordingly to make the power domain core be aware of it.

Reviewed-by: Frank Li <frank.li@nxp.com>
Reviewed-by: Ranjani Vaidyanathan <Ranjani.Vaidyanathan@nxp.com>
Signed-off-by: Dong Aisheng <aisheng.dong@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/soc/imx/pm-domains.c |    7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/soc/imx/pm-domains.c b/drivers/soc/imx/pm-domains.c
index b87272f..aa99a18 100644
--- a/drivers/soc/imx/pm-domains.c
+++ b/drivers/soc/imx/pm-domains.c
@@ -279,6 +279,13 @@ static int __init imx8_add_pm_domains(struct device_node *parent,
 			imx8_pd->pd.attach_dev = imx8_attach_dev;
 			imx8_pd->pd.detach_dev = imx8_detach_dev;
 
+			imx8_pd->pd.states[0].power_off_latency_ns = 25000;
+			imx8_pd->pd.states[0].power_on_latency_ns =  25000;
+			imx8_pd->pd.states[1].power_off_latency_ns = 2500000;
+			imx8_pd->pd.states[1].power_on_latency_ns =  2500000;
+
+			imx8_pd->pd.state_count = 2;
+
 			if (of_property_read_bool(np, "early_power_on")
 				&& index < (sizeof(early_power_on_rsrc) /
 				sizeof(sc_rsrc_t))) {
-- 
1.7.9.5

